rust/arrow/src/compute/kernels/sort.rs

```diff
@@ -668,4 +830,300 @@ mod tests {
             ],
         );
     }
+
+    #[test]
+    fn test_lex_sort_single_column() {
+        let input = vec![SortColumn {
+            values: Arc::new(PrimitiveArray::<Int64Type>::from(vec![
+                Some(17),
```

Comment:
Could you add `f64::NAN` to the values being tested.

Reply:
@andygrove this PR currently doesn't support float array due to `Ord` trait constraints. I recommend adding float support as a follow up after we finalize the design in https://github.com/apache/arrow/pull/7193.

Reply:
I got stuck while working on #7193, there'll be merge conflicts between that and this PR, so if this one gets merged first, I'll attempt adding `f64;NAN` when I rebase